flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/ttl/RocksDbTtlCompactFiltersManager.java

```diff
@@ -125,11 +144,23 @@ public void configCompactFilter(
             ColumnFamilyOptions columnFamilyOptions =
                     columnFamilyOptionsMap.get(stateDesc.getName());
             Preconditions.checkNotNull(columnFamilyOptions);
-            columnFamilyOptions.setPeriodicCompactionSeconds(
-                    rocksdbCompactFilterCleanupStrategy.getPeriodicCompactionTime().getSeconds());
+
+            Duration periodicCompactionTime =
+                    rocksdbCompactFilterCleanupStrategy.getPeriodicCompactionTime();
+            if (periodicCompactionTime == null) {
+                periodicCompactionTime = this.periodicCompactionTime;
+            }
+            if (periodicCompactionTime != null) {
+                columnFamilyOptions.setPeriodicCompactionSeconds(
+                        periodicCompactionTime.getSeconds());
+            }
 
             long queryTimeAfterNumEntries =
                     rocksdbCompactFilterCleanupStrategy.getQueryTimeAfterNumEntries();
+            queryTimeAfterNumEntries =
+                    (queryTimeAfterNumEntries > 0)
+                            ? queryTimeAfterNumEntries
+                            : this.queryTimeAfterNumEntries;
```

Comment:
User could specify `StateTtlConfig` for each state, which should take the precedence over configuration value. I suggest making `getRocksdbCompactFilterCleanupStrategy` only consider the user-specified value and removing the default, by this way to determine whether user have set value in `StateTtlConfig` instead of using `queryTimeAfterNumEntries > 0`. Cause `queryTimeAfterNumEntries <= 0` gives a meaning that it is disabled.

Reply:
Thank you for reviewing. I will make the necessary modifications.